qubes/api/admin.py

```diff
@@ -336,6 +354,49 @@ def vm_volume_import(self):
 
         return '{} {}'.format(size, path)
 
+    @qubes.api.method('admin.vm.tag.List', no_payload=True)
+    @asyncio.coroutine
+    def vm_tag_list(self):
+        assert not self.arg
+
+        tags = self.dest.tags
+
+        tags = self.fire_event_for_filter(tags)
+
+        return ''.join('{}\n'.format(tag) for tag in sorted(tags))
+
+    @qubes.api.method('admin.vm.tag.Get', no_payload=True)
+    @asyncio.coroutine
+    def vm_tag_get(self):
+        qubes.vm.Tags.validate_tag(self.arg)
+
+        self.fire_event_for_permission()
+
+        return '1' if self.arg in self.dest.tags else '0'
```

Comment:
As I said earlier, I'd like this one to say 
```python
return self.arg if self.arg in self.dest.tags else ''
```

The only string that's `False` is an empty string. As to the value, we could either use `'1'`, since it's non-empty and clearly "true", or full tag name.

Do we allow zero-length tag name?